src/hooks/use-fetch-posts/use-fetch-posts.js

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+export const fetchPosts = async (
+  subreddit,
+  postAmount = 500,
+  lastPostId,
+  fetchedPosts,
+) => {
+  let postId = typeof (lastPostId) === 'undefined' ? null : lastPostId;
+  let posts = typeof (fetchedPosts) === 'undefined' ? [] : fetchedPosts;
+  let noMorePosts = false;
+
+  const response = await fetch(`https://www.reddit.com/r/${subreddit}/top.json?t=year&limit=100&after=${postId}`);
+
+  if (!response.ok) return posts;
+
+  const { data: { children, after } } = await response.json();
+
+  postId = after;
+  posts = posts.concat(children);
+  noMorePosts = children.length < 100; // based on `limit` parameter in url
+
+  if (posts.length >= postAmount || noMorePosts) return posts.slice(0, postAmount);
+  return fetchPosts(subreddit, postAmount, postId, posts);
```

Comment:
If you use `after` here instead of `postId` you can get rid of the following line.

```
let postId = typeof (lastPostId) === 'undefined' ? null : lastPostId;
```

You can then initialize the param like `lastPostId = null`. No param reassignment necessary

Reply:
Good catch :smile: